src/Util.h

```diff
@@ -220,85 +220,6 @@ std::vector<T*> toPtrVector(const std::vector<std::unique_ptr<T>>& v) {
     return res;
 }
 
-/**
- * A utility function enabling the creation of a vector of pointers.
- */
-template <typename T>
-std::vector<const T*> toConstPtrVector(const std::vector<std::unique_ptr<T>>& v) {
```

Comment:
I hadn't seen this before. Perhaps we should be using it more? We use the non-const version in several const methods, so this does seem preferable.

Reply:
I would suggest to re-instantiate this template if we want to use const versions for methods like getChildNodes(); I feel this would be the only use-case. 

However, if we change this we need to revisit all the visit/visitor code templates. That might be a pull-request on its own (and lengthy).

Reply:
The current philosophy is that what is not used must go out.